src/hammer-vlsi/cli_driver_test.py

```diff
@@ -39,6 +42,199 @@ def generate_dummy_config(self, syn_rundir: str, config_path: str, top_module: s
+
+        # Generate a config for testing.
+        top_module = "dummy"
+        config_path = os.path.join(syn_rundir, "run_config.json")
+        log_path = os.path.join(syn_rundir, "log.txt")
+        self.generate_dummy_config(syn_rundir, config_path, top_module)
+
+        with HammerLoggingCaptureContext() as c:
+            # Running syn-to-par on a not-output config should fail.
+            with self.assertRaises(SystemExit) as cm:  # type: ignore
+                CLIDriver().main(args=[
+                    "syn-to-par",  # action
+                    "-p", config_path,
+                    "--log", log_path,
+                    "--syn_rundir", syn_rundir,
+                    "--par_rundir", par_rundir
+                ])
+            self.assertEqual(cm.exception.code, 1)
+        self.assertTrue(c.log_contains("Input config does not appear to contain valid synthesis outputs"))
+
+        # Cleanup
+        shutil.rmtree(syn_rundir)
+        shutil.rmtree(par_rundir)
+
+    def test_syn_to_par_same_as_syn_par(self) -> None:
+        """
+        Test that syn-par generates the same par input as calling syn,
+        syn-to-par.
+        """
+
+        # Set up some temporary folders for the unit test.
+        syn_rundir = tempfile.mkdtemp()
+        par_rundir = tempfile.mkdtemp()
+
+        # Generate a config for testing.
+        top_module = "dummy"
+        config_path = os.path.join(syn_rundir, "run_config.json")
+        syn_out_path = os.path.join(syn_rundir, "syn_out.json")
+        syn_to_par_out_path = os.path.join(syn_rundir, "syn_par_out.json")
+        self.generate_dummy_config(syn_rundir, config_path, top_module)
+
+        # Run syn-to-par
+        self.run_syn_to_par_with_output(config_path, syn_rundir, par_rundir,
+                                        syn_out_path, syn_to_par_out_path)
+
+        # Run syn-par
+        with self.assertRaises(SystemExit) as cm:  # type: ignore
+            CLIDriver().main(args=[
+                "syn-par", # action
+                "-p", config_path,
+                "--syn_rundir", syn_rundir,
+                "--par_rundir", par_rundir
+            ])
+        self.assertEqual(cm.exception.code, 0)
+
+        # Check that the syn-to-par generated par input and the syn-par
+        # generated par input are the same, modulo any synthesis.outputs.*
+        # settings since they don't matter for par input.
+        with open(syn_to_par_out_path, "r") as f:
+            with open(os.path.join(syn_rundir, "par-input.json"), "r") as f2:
+                syn_to_par_output = json.loads(f.read())
+                syn_to_par_output = dict(filter(lambda i: "synthesis.outputs." not in i[0], syn_to_par_output.items()))
+                syn_par_output = json.loads(f2.read())
+                syn_par_output = dict(filter(lambda i: "synthesis.outputs." not in i[0], syn_par_output.items()))
+                self.assertEqual(syn_to_par_output, syn_par_output)
```

Comment:
What is expected to differ between the non-filtered versions? Do we even want the filter?

Reply:
The filter simply removes `synthesis.outputs.*` settings. The reason there is a discrepancy is because `syn-par` takes the output-only result from synthesis and combines it with the original (synthesis.output free config), whereas `syn-to-par` combines it with the synthesis output which includes "synthesis.output.*" settings. Of course we could hack/special-case to `syn-to-par` but it seems like it doesn't really matter and avoids an unnecessary hack.

Reply:
Yeah its fine to leave as is. Thanks for the explanation.